File: examples/sdk/node/discover.js
import { Client, DiscoveryFilter } from "../../../bindings/node/index.js";

const client = new Client("http://127.0.0.1:8765", "dev-token");

try {
  const nodes = client.discover(
    new DiscoveryFilter("office-a").withService("_demo._tcp").addTag("stable"),
  );
  console.log(nodes);
} finally {
  client.close();
}

File: examples/sdk/python/discover.py
from lnd import Client, DiscoveryFilter


with Client("http://127.0.0.1:8765", "dev-token") as client:
    nodes = client.discover_with_auto_scope_overlap(
        DiscoveryFilter().with_discovery_domain("office-a").with_service("_http._tcp").add_tag("stable")
    )
    print(nodes)
